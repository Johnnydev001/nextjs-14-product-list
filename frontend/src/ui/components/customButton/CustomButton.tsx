'use client';
import React from "react";

export const CustomButton = ({disabled= false, onClick = () => {}, className= '', children = null}:  {disabled?: boolean,onClick: React.MouseEventHandler<HTMLButtonElement> , className?: string, children?: any}) => {
    return (<button disabled={disabled} onClick={onClick} role={'button'} aria-label={'Custom generic button to be used everywhere'} className={className}>
        {children}
        </button>)
}