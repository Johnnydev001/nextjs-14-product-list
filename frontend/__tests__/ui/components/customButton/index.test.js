import { render, fireEvent } from '@testing-library/react';
import {CustomButton} from "../../../../src/ui/components/customButton/CustomButton";

describe('Custom button component tests', () => {
    it('should render the custom button component with default props and match the snapshot', () => {
        const {container, getByText } = render(<CustomButton onClick={jest.fn()}/>)

        expect(container).toMatchSnapshot()
    })

    it('should render the custom button component with custom props and match the snapshot', () => {
        const props = {
            onClick: jest.fn(),
            children: (<div>Button text</div>),
            className: 'text-lg'
        }
        const {container, getByRole } = render(<CustomButton {...props}/>)

        expect(getByRole('button')).toHaveClass('text-lg')

        expect(container).toMatchSnapshot()
    })

    it('should handle the button click', () => {
        const props = {
            onClick: jest.fn(),
            children: (<div>Button text</div>),
            className: 'text-lg'
        }
        const {getByText } = render(<CustomButton {...props}/>)

        const button = getByText('Button text')

        fireEvent.click(button)

        expect(props.onClick).toHaveBeenCalled()
    })
})