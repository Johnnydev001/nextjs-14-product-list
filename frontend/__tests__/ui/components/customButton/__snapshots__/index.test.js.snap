// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom button component tests should render the custom button component with custom props 1`] = `
<div>
  <button
    aria-label="Custom generic button to be used everywhere"
    class="text-lg"
    role="button"
  >
    <div>
      Button text
    </div>
  </button>
</div>
`;

exports[`Custom button component tests should render the custom button component with custom props and match the snapshot 1`] = `
<div>
  <button
    aria-label="Custom generic button to be used everywhere"
    class="text-lg"
    role="button"
  >
    <div>
      Button text
    </div>
  </button>
</div>
`;

exports[`Custom button component tests should render the custom button component with default props 1`] = `
<div>
  <button
    aria-label="Custom generic button to be used everywhere"
    class=""
    role="button"
  />
</div>
`;

exports[`Custom button component tests should render the custom button component with default props and match the snapshot 1`] = `
<div>
  <button
    aria-label="Custom generic button to be used everywhere"
    class=""
    role="button"
  />
</div>
`;
